compiler-rt/lib/rtsan/rtsan_interceptors_posix.cpp

```diff
@@ -1021,6 +1021,16 @@ INTERCEPTOR(int, setsockopt, int socket, int level, int option,
 #define RTSAN_MAYBE_INTERCEPT_SETSOCKOPT
 #endif
 
+#if !SANITIZER_APPLE
```

Comment:
Apple has socketpair, for the record (or at least mine does)

```
> man 2 socketpair | head
SOCKETPAIR(2)                 System Calls Manual                SOCKETPAIR(2)

NAME
     socketpair – create a pair of connected sockets

SYNOPSIS
     #include <sys/socket.h>

     int
     socketpair(int domain, int type, int protocol, int socket_vector[2]);
```

Reply:
Yes it does I see however only netbsd and linux intercept it in sanitizer common so I was wary here.

Reply:
Looking at `tsan_interceptors_posix.cpp` and `msan_interceptors.cpp`, it seems like socketpair is unguarded (aka available on everything including OSX).

Do you have a mac machine to test it out? If not, let me know and I'd be happy to test it on mine

Reply:
fair point ; testing it now.